src/components/Form/Form.js

```diff
@@ -1,9 +1,42 @@
-import React from "react";
+import React, { Component } from 'react';
 import './Form.css'
 
-const Form = () => {
+class Form extends Component {
+    constructor() {
+        super()
+        this.state = {
+            zip: ''
+        }
+    }
+
+    handleChange = (event) => {
+        this.setState({
+            [event.target.name] : event.target.value
+        })
+    }
+
+    render() {
     return (
-        <form>Form</form>
+        <form onSubmit={(event) => this.props.changeZipCode(this.state.zip, event)}>
+            <p>Zip Code:</p>
+            <input
+                className='zip-code-entry'
+                type='text'
+                data-cy='zip-code-entry'
+                value={this.state.zip}
+                name='zip'
+                placeholder='Enter Zip Code'
+                pattern='[0-9]{5}'
```

Comment:
WHOA a pattern attribute to make sure zip code is 5 characters long for error handling!! 😎